sdk/communication/communication-sms/test/internal/node/smsClientWithToken.internal.spec.ts

```diff
@@ -1,26 +1,37 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { env, isPlaybackMode, record, Recorder } from "@azure/test-utils-recorder";
-import { SmsSendRequest, SmsClient } from "../src/smsClient";
+/**
+ *  These are a duplicate of the public SmsClientWithToken.spec.ts tests, but with additional logic to enable recording/playback.
+ *  This is a workaround because Http Requests with Randomized UUIDs do not play well with the recorder
+ *  These tests will be skipped in Live Mode since the public tests run in live mode only.
+ */
+
+import { env, isLiveMode, isPlaybackMode, record, Recorder } from "@azure/test-utils-recorder";
+import { SmsSendRequest, SmsClient } from "../../../src/smsClient";
 import { assert } from "chai";
 import { isNode } from "@azure/core-http";
 import * as dotenv from "dotenv";
 import * as sinon from "sinon";
 import { parseConnectionString } from "@azure/communication-common";
-import { createCredential, recorderConfiguration } from "./utils/recordedClient";
-import { Uuid } from "../src/utils/uuid";
+import { createCredential, recorderConfiguration } from "../../utils/recordedClient";
+import { Uuid } from "../../../src/utils/uuid";
 import { Context } from "mocha";
 
 if (isNode) {
   dotenv.config();
 }
 
-describe("SmsClientWithToken [Playback/Live]", async () => {
+describe("SmsClientWithToken [Playback/Record]", async () => {
```

Comment:
Instead of duplicating everything in internal, I wonder if we can export the Mocha suites from public and then import them in internal?

https://alxolr.com/articles/how-to-separate-mocha-tests-in-multiple-files

Reply:
looking at this now